### ISSUE TITLE
Any progress on implementing CRF RNN in tensorflow?

### ISSUE THREAD

**Issue by gaopeng-eugene** [jasonbunk/tensorflow-bilateral-permutohedral#1]
The receptive field of bilateral filter is the whole image. Do you have any suggestion on implementing an window bilateral filter.

**Comment by gaopeng-eugene** [comment]
Where is the describe function in your test code?

**Comment by gaopeng-eugene** [comment]
can you provide the trained_segment_weights.pkl file?

**Comment by jasonbunk** [comment]
About the receptive field, in theory a Gaussian has an infinite receptive field, but in the algorithm it is approximated by a series of filters on a lattice that are only 3 lattice points wide (the filter coefficients are just 1,2,1 on each lattice axis). You can change the spatial window size with the argument "stdv_space". Details about the lattice are in the paper: [http://graphics.stanford.edu/papers/permutohedral/permutohedral.pdf](http://graphics.stanford.edu/papers/permutohedral/permutohedral.pdf)

About the describe function, I added it to test_utils, it just prints debug info about a variable.

**Comment by jasonbunk** [comment]
The trained_segment_weights.pkl file is generated by test/test_segment.py after letting it train and keyboard-interrupting it to stop the training. I pushed a commit with the describe function.
